extensions/security/runtime/src/main/java/io/quarkus/security/runtime/interceptor/check/AuthenticatedCheck.java

```diff
@@ -15,8 +19,23 @@ private AuthenticatedCheck() {
 
     @Override
     public void apply(SecurityIdentity identity, Method method, Object[] parameters) {
+        if (isAuthorizationDisabled()) {
+            return;
+        }
         if (identity.isAnonymous()) {
             throw new UnauthorizedException();
         }
     }
+
+    private boolean isAuthorizationDisabled() {
+        ArcContainer container = Arc.container();
+        if ((container == null) || !container.isRunning()) {
+            return false;
+        }
+        InstanceHandle<AuthorizationController> instance = container.instance(AuthorizationController.class);
```

Comment:
This is relatively slow, as it does bean resolution on every request. Can we cache the AuthorizationController in a volatile field instead?:

Reply:
Good idea. Fixed